### DISCUSSION TITLE
Quarkus at J4K

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#27794] [Events]
J4K 2022
A Conference for Java Developers on Kubernetes.
J4K is an industry leading conference that combines the best of open source and middleware communities for developing applications on Kubernetes.
Containers and Kubernetes have been gaining serious traction in the market within the last year and organizations are now exploring and leveraging the benefits of hybrid and multi-cloud environments. We are at a convergence point where middleware and application services need to be made efficient to run on containers and Kubernetes.

This conference is targeted to developers and industry influencers of middleware and application services on Kubernetes.

Event type: Virtual
Date: September 28, 2022
https://www.j4k.io/

Register at: https://whova.com/portal/registration/jkhyb_202210/

Quarkus Sessions

**Integration with Camel Quarkus in the cloud era**
Presenter: Alexandre Gallice
Time: 11:00 am - 11:35 am EDT

Session Description:
Apache Camel is the proven integration swiss knife for years. In today's world of workloads moving to the cloud, the need for disparate systems to communicate remains more than ever. This context makes a Kubernetes Java stack like Quarkus a good fit to implement Camel routes.

In this session, the attendance can first expect a quick reminder about Camel Quarkus basics. Further, a concrete demo will illustrate the interest of running Camel Quarkus on top of Knative.

**From Zero to Hero in Kubernetes Native Java**
Presenter: Daniel Oh
Time: 1:30 pm - 2:05 pm EDT

Session Description: 

**Quarkus for Spring Developers**
Presenter: Eric Deandrea
Time: 2:15 pm - 2:50 pm EDT

Session Description:
Are you familiar with Spring and like learning new things?

In this session, I will show concepts and conventions familiar to Spring developers and how those same concepts and conventions can be implemented in Quarkus, all while highlighting similarities and differences between them. Additionally, I will show similarities and differences in how testing is done, highlighting Quarkus Dev Services and continuous testing.

This session will be mostly live coding while minimizing the amount of slides. I will introduce an existing Spring application with a full test suite and build a Quarkus equivalent version of it, live. Content will come from my book, "[Quarkus for Spring Developers](https://red.ht/quarkus-spring-devs)."

**Practices and developer tools for cloud, container and K8s**
Presenter: Eric Deandrea
Time: 11:00 AM-11:35 AM

Session Description:
Everyone loves writing tests, don’t they? How do you write good tests? What tools are available for you to write good tests?

In this session, I will dive into the many features of Quarkus that help developers write good tests. I will highlight some of the features of Quarkus, Dev Services and Continuous Testing, which help make testing easier. Additionally, I will live-code some tests for common use cases developers encounter, including unit, integration, and “black box” testing of imperative and reactive RESTful and event-driven applications that use common services, such as databases and Kafka brokers. I will discuss techniques such as mocking, spying, and interaction-based testing/verification.

**Enter Serverless Functions Journey with Quarkus**
Presenter: Daniel Oh
Time: 3:45 pm - 4:20 pm EDT

Session Description: 
This workshop is designed to be a hands-on experience on how developers can build agnostic serverless Java functions using Quarkus, a new Kubernetes native Java framework with familiar tools (e.g. CLI, Maven, Gradel). The workshop covers deploying service functions to AWS Lambda, optimizing the functions, making them portable across multiple serverless platforms ( e.g., AWS Lambda, Azure Function, Google Cloud Platform, Kubernetes Knative), and using the latest handy command-line tool (e.g. Kn func) to enable a Buildpack for function development & deployment in minutes. Lab participants will be provided a free sandbox for serverless deployments.

Here are the lab instructions (WIP) - https://github.com/RedHat-Middleware-Workshops/enter-serverless-functions/blob/main/README.md

**Comment by edeandrea** [comment]
Also - How about some tests with your applications - 11am - 11:35am

Everyone loves writing tests, don’t they? How do you write good tests? What tools are available for you to write good tests?

In this session, I will dive into the many features of Quarkus that help developers write good tests. I will highlight some of the features of Quarkus, Dev Services and Continuous Testing, which help make testing easier. Additionally, I will live-code some tests for common use cases developers encounter, including unit, integration, and “black box” testing of imperative and reactive RESTful and event-driven applications that use common services, such as databases and Kafka brokers. I will discuss techniques such as mocking, spying, and interaction-based testing/verification.

**Reply by insectengine** [reply]
Added it to the original list.
